### ISSUE TITLE
Release plans

### ISSUE THREAD

**Issue by fingolfin** [libsdl-org/SDL_mixer#369]
As far as I can tell, the last release was 2018 and there have been > 300 commits since then. Any chance for a new release? Or perhaps there are blockers for that?

**Comment by slouken** [comment]
No blockers, it's on the TODO list. :)

**Comment by a-hurst** [comment]
@slouken Is there a checklist of issues/PRs to address before the next release, like there is for the main SDL2 project? Would be good to know if there's anything we can do to help with the new release, I'm looking forward to there being official code-signed Universal `SDL_image` binaries for macOS!

**Comment by slouken** [comment]
The Xcode project needs to be updated for iOS and tvOS, and the third party dependencies need to be built as included frameworks for macOS, iOS, and tvOS.

**Comment by sezero** [comment]
3rd party deps are already built & included as bundled frameworks for macOS (not signed though). Can't do anything about iOS and tvOS.

**Comment by a-hurst** [comment]
@slouken Do you mean the FLAC/modplug/mpg123/etc. frameworks need to be compiled for iOS/tvOS and included in the source, the way they are for macOS in `./Xcode/Frameworks/`? From what I can tell the iOS and tvOS projects currently build those as part of the `SDL_mixer` Xcode project itself instead of bundling pre-built versions. It also seems to have already been updated to use the latest versions of the dependencies in `./external` (well, the ones it supports at all: it doesn't seem to build opus or mpg123 support).

I have a number of iOS devices ranging back to iOS 5 and an Apple Silicon Mac (for native ARM Xcode simulators), is there anything I can do to help out here?

**Comment by slouken** [comment]
I've updated the Xcode project to build all the dependencies. I have some more testing to do, but you can look at my recent commits to see what I've been doing. The idea is that all the dependent frameworks are optional, and you can include any of them for formats you want to support and leave the rest out.

**Comment by sezero** [comment]
I think we should fix at least these issues before making a release:
https://github.com/libsdl-org/SDL_mixer/issues/293
https://github.com/libsdl-org/SDL_mixer/issues/290#issuecomment-803624462
https://github.com/libsdl-org/SDL_mixer/issues/281
https://github.com/libsdl-org/SDL_mixer/issues/260
https://github.com/libsdl-org/SDL_mixer/issues/245
Maybe these?
https://github.com/libsdl-org/SDL_mixer/issues/258#issuecomment-931260899
https://github.com/libsdl-org/SDL_mixer/issues/284
And maybe this would be a good feature:
https://github.com/libsdl-org/SDL_mixer/issues/377

**Comment by smcv** [comment]
> I think we should fix at least these issues before making a release

Obviously it would be ideal if a new release fixed all known bugs, but that's not the choice that's available here, unless those bugs have ready-to-merge solutions; the choice is more like whether current SDL_mixer git HEAD is better or worse than 2.0.4.

I think it's also useful to distinguish between regressions (points where git HEAD is actively worse than 2.0.4) and non-regressions (bugs that existed in 2.0.4 and still exist in git HEAD). If a regression isn't fixed, that gives people who are affected by the regression a reason not to want to upgrade, fragmenting the versions in use; but if a non-regression isn't fixed, then that doesn't prevent 2.0.4 -> git HEAD from being an improvement.

**Comment by a-hurst** [comment]
@sezero @slouken Does #383 being merged mean that we're not going to see a new mixer (or image) release until the next version of SDL2 is released with the new version scheme? I've been holding off on a new 2.0.22 release of pysdl2-dll in the off-chance mixer or image were going to get new releases (and crucially for my case, Apple Silicon support), but if that's been pushed back again I'll just push the new release now.

Also, thanks again for all your continued maintenance work on these projects! I work in cognitive science research, and having a thoroughly cross-platform framework for video/audio/inputs/fonts/etc. makes programming and sharing experiments a lot easier.

**Comment by slouken** [comment]
No, the versioning of each library can be released independently. I'm working through regressions and testing for the satellite libraries now.

**Comment by madebr** [comment]
Not that I can think of.
Linux, Windows and Macos are tested on CI.
I tested the VC and MinGW sdk's. So the exotic platforms are the black spot.
